### ISSUE TITLE
Commands will fail if not run somewhere under /home/<user>/*

### ISSUE THREAD

**Issue by EmilienM** [pierreprinetti/oc-setversion#2]
The current code for `version_from_config` is looking for the configuration file from the target directory and escalate until `/`:

```
+ oc get nodes
+ declare -r config_file=/home/emilien/.config/oc-setversion/paths.conf log_level= bin_dir=/home/emilien/.local/bin
+ '[' get == setversion ']'
+ declare version
++ version_from_config
++ '[' -f /home/emilien/.config/oc-setversion/paths.conf ']'
++ declare target= version=
+++ realpath /tmp/performance-addon-operators
++ target=/tmp/performance-addon-operators/x
++ '[' /tmp/performance-addon-operators/x '!=' / ']'
+++ dirname /tmp/performance-addon-operators/x
++ target=/tmp/performance-addon-operators
+++ sed -n '0,\|/tmp/performance-addon-operators |s|/tmp/performance-addon-operators ||p' /home/emilien/.config/oc-setversion/paths.conf
++ version=
++ '[' -n '' ']'
++ '[' /tmp/performance-addon-operators '!=' / ']'
+++ dirname /tmp/performance-addon-operators
++ target=/tmp
+++ sed -n '0,\|/tmp |s|/tmp ||p' /home/emilien/.config/oc-setversion/paths.conf
++ version=
++ '[' -n '' ']'
++ '[' /tmp '!=' / ']'
+++ dirname /tmp
++ target=/
+++ sed -n '0,\|/ |s|/ ||p' /home/emilien/.config/oc-setversion/paths.conf
++ version=
++ '[' -n '' ']'
++ '[' / '!=' / ']'
+ export version=
+ version=
+ readonly version
+ '[' -z '' ']'
+ echo 'Error: no configuration set for this directory.'
Error: no configuration set for this directory.
```

**Comment by pierreprinetti** [comment]
> The current code for version_from_config is looking for the configuration file from the target directory and escalate until /

Yes, that is an accurate description of the behaviour of the tool. What is the issue you are facing?

**Comment by pierreprinetti** [comment]
The tool expects (or creates) the configuration file in `${XDG_CONFIG_HOME:-${HOME}/.config}/oc-setversion/paths.conf`.

The tool uses the directory structure to define a context. In a filesystem that looks like this:
```
$ tree
.
├── a
│   └── b
│       └── c
└── d
```

Let's say you have a project in `a/b/`, and to work on it, you need OCP 4.6. Here's how you would set the desired version:

```shell
$ cd a/b/
$ oc setversion latest-4.6
```

At this point, calling `oc` from `a/b/` and its subdirectories (e.g. `a/b/c/`) will result in a call to the latest version of OCP 4.6. Calling from `a/` or `d/` would result in an error, because there is no configuration outside `a/b/`.

The directory traversal you are seeing is the result of recursively checking each parent folder looking for one for which you set a version.

One tip I have added to [go-setversion's README](https://github.com/pierreprinetti/go-setversion#tip) (but indeed, not here), is to set the latest version (or the one you generally want) on `/`, so that you never get the error you've seen.

**Comment by pierreprinetti** [comment]
Do you find this behaviour surprising? How did you expect the tool to work instead?

**Comment by EmilienM** [comment]
> How did you expect the tool to work instead?

- binaries paths are in `${XDG_CONFIG_HOME:-${HOME}/.config}/oc-setversion/paths.conf`
- the tool is able to figure out where to find the config (since it's a static paths.conf) and therefore find the binaries, so I can call the binary from anywhere in the tree.

**Comment by pierreprinetti** [comment]
I work a lot from `/tmp`, and the tool works! Can you give me an example, or ideally a reproducer, to understand where the issue is?

**Comment by pierreprinetti** [comment]
Oh wait, I read your first post again, and I now understand where the issue is.

In my configuration file (which sits in `$HOME/.config/oc-setversion/paths.conf`), I have a _default_ value set, under `/`. For example:

```plaintext
/ 4.12.5
/var/home/pierre/ocp 4.12.7
```

The first line being set to `/` acts as a catch-all for all paths where I didn't set an override.

Does that help, @EmilienM ?
